File: java/src/main/java/publicdomain/test/snippets/system/SystemTime.java
package publicdomain.test.snippets.system;

import java.time.Instant;

/**
 * {@link SystemTime} provides date/time values from the system.
 * <p>
 * In case {@link Time} has only one method, instead of using this class, you could also use a method reference of
 * {@link Instant#now()}.
 */
public class SystemTime implements Time {
    @Override
    public Instant now() {
        return Instant.now();
    }
}

File: dotnet/TestSnippets.ForProduction/System/SystemTime.cs
using System;

namespace PublicDomain.TestSnippets.System
{
    /// <summary>
    /// <see cref="SystemTime"/> provides date/time values from the system.
    /// </summary>
    public class SystemTime: ITime
    {
        /// <inheritdoc cref="ITime"/>
        public DateTime Now => DateTime.UtcNow;
    }
}